### ISSUE TITLE
Reference documentation lists beans that are not part of the public API by default

### ISSUE THREAD

**Issue by vincentditlevinz** [spring-projects-experimental/spring-modulith#187]
This kind of repository doesn't appear in the generated documentation:

```
public interface SkillRepository extends ReactiveSortingRepository<Skill, Integer> {}
```

I tried to add `@Repository` without luck. 

More generally it seems only concrete classes implementing an interface and annotated with @Component, @Service, @Repository are depicted, for instance:

```
@Service
public interface SkillServiceImpl implements SkillService {
....
}
```

Note that the same implementation, with @Service annotation but without extendind the SkillService interface is not depicted too.

**Comment by odrotbohm** [comment]
Do you have a reproducer for that? I see Spring Data repositories detected and rendered properly in quite a few production deployments. Also, I have just added unit tests for their detection as `ArchitecturallyEvidentType`.

**Comment by vincentditlevinz** [comment]
I am on holidays but I am very interested by this project so I've adapted a project od mine that [reproduces the issue](https://github.com/vincentditlevinz/api-domains-issue). :smile:

**Comment by odrotbohm** [comment]
Thanks for the reproducer. Even more so, since you're on vacation. 🙃 This essentially works as designed. The Application Module Canvas describes the *exposed* elements of a module. As both the service implementation and the repository interface are located in [implementation packages](https://docs.spring.io/spring-modulith/docs/current/reference/html/#fundamentals.modules.advanced), they're inaccessible to other modules of the system, they do not appear in the canvas by default.

There are two ways to conquer the problem:
- If you simply would like to reveal such components in the reference documentation, trigger the documentation creation with `CanvasOptions.revealInternals()`. This changes the documentation but still doesn't allow other modules to access the internal components.
- Alternatively, you can expose the individual packages or types by annotating them with `@NamedInterface`. This will make them part of the API of the module which causes them to be included in the default canvas but also accessible to components in other modules.

**Comment by vincentditlevinz** [comment]
Thanks for this thorough explanation, I totally missed the
CanvasOptions.revealInternals() option.
Just one remaining question: SkillService is in the services subdirectory,
it should be depicted as an internal service too, shouldn't it ? Why does
it appear in documentation when I use the default canvas config then ?
Don't know if it's a bug, but there is something that is not very intuitive
for me here.

Le sam. 29 avr. 2023 à 23:02, Oliver Drotbohm ***@***.***> a
écrit :

**Comment by odrotbohm** [comment]
That is indeed a bug. The interface should only be listed if it is part of the exposed API. It looks like currently the latter constraint is missing. Investigating.

**Comment by odrotbohm** [comment]
This should be in place. I've tried a local build against your project, and it now does not list the service interface and implementation by default. Feel free to give the snapshots a try.

**Comment by vincentditlevinz** [comment]
Thanks, indeed, the issue is fixed in 0.6.0-SNAPSHOT :+1:
